Add Term Entry: Python String Method `.join()` (#601)
Codecademy/docs: content/python/concepts/strings/terms/join/join.md

---
Title: '.join()'
Description: 'Concatenates all items from an iterable into a single string.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The `.join()` method concatenates all items from an iterable into a single string.

## Syntax

> 

Reviewer: ```suggestion

The `.join()` method is called on by being appended to a `separator` string:

```

Author: @Dusch4593 I like this sentence but I feel like "appended" here is forced. Can we just write it as, "The `.join()` method is called on a `separator` string:" and find another place for "append".

Author: @Dusch4593 I like this sentence but I feel like "appended" here is forced. Can we just write it as, "The `.join()` method is called on a `separator` string:" and find another place for "append".

```pseudo

> string.join(iterable)

Reviewer: ```suggestion
separator.join(iterable)
```

```

> - `string` is the separator `.join()` is called on. The separator is used to separate each element of the iterable. It can be any string including an empty string.

Reviewer 2: ```suggestion
- The `.join()` method is called on a separator of type string. The separator can be any string, including an empty string, and is used to separate each element of the iterable.
```

I think it's important to make some distinction that the `string` mentioned in the pseudo block represents any string. This is why I'm suggesting taking away the backticks.

> - The `.join()` method is called on a separator of type string. The separator can be any string, including an empty string, and is used to separate each element of the iterable.

Reviewer: ```suggestion
- The `separator` can be any string, even an empty one, and is placed between each element from the `iterable`.
```

Reviewer 2: This makes more sense! One bullet point for the separator and a second point for the iterable 👍 agreed

Reviewer: So change `separator` and `iterable` to just separator and iterable?

Reviewer 2: This is very "nitty-gritty."

Notice how [.replace()](https://www.codecademy.com/resources/docs/python/strings/replace) and [.index()](https://www.codecademy.com/resources/docs/python/strings/index) wrap the argument placeholders in backticks, but for [.partition()](https://www.codecademy.com/resources/docs/python/strings/partition), the keyword is not wrapped in backticks.

A similar issue in [PR 537 file](https://github.com/Codecademy/docs/pull/537/files).

So maybe for these sentence/paragraph list items, we'll do just as you said which is to change `separator` and `iterable` to separator and iterable

Author: @yangc95 maybe down the line we can have a consensus on this. I personally like using backticks around these fields only because we don't get to use many emphasis features like italics or bold.

Author: @yangc95 @Dusch4593 But to be clear, we are removing the backticks, right?

Reviewer 2: Sorry, I keep going back and forth. Until there's a consensus on this, let's keep the backticks for everything except "string"

Author: ok 😂. I'll just commit everything so we can see how it would look. We can always change it again later.

Reviewer 2: 🥲 😆 Sounds good

> - `iterable` is any object that can be iterated over like tuples, lists, etc. All values of the iterable must be strings.

Reviewer 2: ```suggestion
- The `iterable` is any object that can be iterated over like tuples or lists. All values of the `iterable` must be strings.
```

---
Title: '.join()'
Description: 'Concatenates all items from an iterable into a single string.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'

>   - 'Data Science'

Reviewer: Should we could part with one of these and keep two for the `Subjects`?

Author: @Dusch4593  Which one do you want to remove? I'm fine with any.

Reviewer 2: @Dusch4593 were you thinking of parting with 'Code Foundations'?

Reviewer: I was actually thinking maybe "Computer Science" since Python is more data science-y (right?) and strings are sort of related to code foundations(?)

Author: So I took a quick look, currently, the subjects listed for [Python Strings](https://www.codecademy.com/resources/docs/python/strings) are 'Code Foundations', 'Computer Science', 'Data Science'. 

Every string term entry, however, have only  'Data Science' & 'Computer Science' listed

Reviewer: Ah, well if that's the case then yeah we should do away with "Code Foundations". 😄

Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The `.join()` method concatenates all items from an iterable into a single string.

## Syntax

```pseudo
string.join(iterable)
```

- `string` is the separator `.join()` is called on. The separator is used to separate each element of the iterable. It can be any string including an empty string.
- `iterable` is any object that can be iterated over like tuples, lists, etc. All values of the iterable must be strings.

## Example

> The following combines the elements of a list using a space (`" "`) as a separator:

Reviewer: ```suggestion
The following example returns a string version of the `fruits` list, separated by a single space (`" "`):
```

```py
fruits = ["Apples", "Bananas", "Blueberries"]

combined = " ".join(fruits)
print(combined)

# Output: Apples Bananas Blueberries
```

## Codebyte Example

Using `.join()` to combine elements of a tuple with a dash (`"-"`) as a separator:

```codebyte/python
vehicles = ("bicycle", "car", "moped", "truck")

joined = "-".join(vehicles)
print(joined)

> 
> # Output: bicycle-car-moped-truck

Reviewer: ```suggestion
```
